app/controllers/assignment_invitations_controller.rb

```diff
@@ -2,16 +2,29 @@
 
 class AssignmentInvitationsController < ApplicationController
   include InvitationsControllerMethods
+  include SetupRepo
 
   before_action :check_user_not_previous_acceptee, only: [:show]
-  before_action :ensure_submission_repository_exists, only: [:success]
+  before_action :ensure_submission_repository_exists, only: %i[setup setup_progress success]
+  before_action :check_authorized_repo_setup, only: %i[setup setup_progress]
 
   def accept
     create_submission do
-      redirect_to success_assignment_invitation_path
+      redirect_to setup_assignment_invitation_path
     end
   end
 
+  def setup
+    starter_code_repo_id = current_submission.assignment.starter_code_repo_id
```

Comment:
Small style comment: Reading this code makes you think we need the id for something. Really, we need to know whether or not the id exists for something. We could change this line to:
```
starter_code_repo_present? = current_submission.assignment.starter_code_repo_id.present?
```
which I think reads a little better.